syntax/go.vim

```diff
@@ -166,11 +166,15 @@ syn match       goDecimalInt        "\<-\=\d\+\%([Ee][-+]\=\d\+\)\=\>"
 syn match       goHexadecimalInt    "\<-\=0[xX]\x\+\>"
 syn match       goOctalInt          "\<-\=0\o\+\>"
 syn match       goOctalError        "\<-\=0\o*[89]\d*\>"
+syn match       goBinaryInt         "\<-\=0[bB][01]\+\>"
+syn match       goBinaryError       "\<-\=0[bB][01]*[2-9]\+[01]*\>"
```

Comment:
Why does this match leading and trailing 0 or 1? Would `<-\=0[bB].*[^01]` be simpler and catch all the problems?

Reply:
Your regex does not match errors like `0b2201` AFAICT.

Reply:
It does match `0b2201`: https://play.golang.org/p/QUm4VUWnF0e